File: Dot examples/ASP Book Examples/Book Examples/Chapter15/CS/App_Code/Inheritance.cs

public class BaseProduct
{
    private decimal _price;

    public decimal Price
    {
        get { return _price; }
        set { _price = value; }
    }

}

public class ComputerProduct : BaseProduct
{
    private string _processor;

    public string Processor
    {
        get { return _processor; }
        set { _processor = value; }
    }

}

public class TelevisionProduct : BaseProduct
{

    private bool _isHDTV;

    public bool IsHDTV
    {
        get { return _isHDTV; }
        set { _isHDTV = value; }
    }

}

File: Dot examples/ASP Book Examples/Book Examples/Chapter15/VB/App_Code/Inheritance.vb

Public Class BaseProduct

    Private _price As Decimal

    Public Property Price() As Decimal
        Get
            Return _price
        End Get
        Set(ByVal Value As Decimal)
            _price = Value
        End Set
    End Property

End Class

Public Class ComputerProduct
    Inherits BaseProduct

    Private _processor As String

    Public Property Processor() As String
        Get
            Return _processor
        End Get
        Set(ByVal Value As String)
            _processor = value
        End Set
    End Property

End Class

Public Class TelevisionProduct
    Inherits BaseProduct

    Private _isHDTV As Boolean

    Public Property IsHDTV() As Boolean
        Get
            Return _isHDTV
        End Get
        Set(ByVal Value As Boolean)
            _isHDTV = value
        End Set
    End Property

End Class

